sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -1278,6 +1306,60 @@ public boolean isTracingEnabled() {
     return getTracesSampleRate() != null || getTracesSampler() != null;
   }
 
+  /**
+   * Returns the list of exception classes that once captured will not be sent to Sentry as {@link
+   * SentryEvent}.
+   *
+   * @return the list of exception classes that once captured will not be sent to Sentry as {@link
+   *     SentryEvent}.
+   */
+  public @Nullable Set<Class<? extends Throwable>> getIgnoredExceptionsForType() {
+    return ignoredExceptionsForType;
+  }
+
+  /**
+   * Sets the list of exception types that once captured will not be sent to Sentry as {@link
+   * SentryEvent}.
+   *
+   * @param ignoredExceptionsForType list of ignored exception types
+   */
+  @SuppressWarnings("unchecked")
+  public void setIgnoredExceptionsForType(final @NotNull Set<Class<?>> ignoredExceptionsForType) {
+    // this method is invoked by Spring Boot integration using reflection when generic information
+    // is erased - thus we need to make sure that only valid throwable types are set.
+    final Set<Class<? extends Throwable>> ignoredExceptionTypes = new HashSet<>();
+    for (final Class<?> clazz : ignoredExceptionsForType) {
+      if (Throwable.class.isAssignableFrom(clazz)) {
+        ignoredExceptionTypes.add((Class<? extends Throwable>) clazz);
+      }
+    }
+    this.ignoredExceptionsForType = ignoredExceptionTypes;
+  }
+
+  /**
+   * Adds exception type to the list of ignored exceptions.
+   *
+   * @param exceptionType - the exception type
+   */
+  public void addIgnoredExceptionForType(final @NotNull Class<? extends Throwable> exceptionType) {
+    if (this.ignoredExceptionsForType == null) {
+      this.ignoredExceptionsForType = new HashSet<>();
+    }
```

Comment:
what's about creating it directly and using a thread-safe `Set` as we do with all the other list/array/map? see eventProcessors and integrations, we already got issues that people were iterating the list and removing from a loop which obviously would throw (if not thread-safe type).